src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -0,0 +1,102 @@
+﻿// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using Dapr.Client;
+using Microsoft.Extensions.Configuration;
+
+namespace Dapr.Extensions.Configuration.DaprSecretStore
+{
+    /// <summary>
+    /// A Dapr Secret Store based <see cref="ConfigurationProvider"/>.
+    /// </summary>
+    internal class DaprSecretStoreConfigurationProvider : ConfigurationProvider
+    {
+        private readonly string store;
+
+        private readonly IEnumerable<DaprSecretDescriptor> secretDescriptors;
+
+        const string localhost = "127.0.0.1";
+
+        static string daprPort => Environment.GetEnvironmentVariable("DAPR_GRPC_PORT") ?? "50001";
+
+        private readonly DaprClient client;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="DaprSecretStoreConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="store">Dapr Secre Store name.</param>
+        /// <param name="secretDescriptors">The secrets to retrieve.</param>
+        /// <param name="client">Dapr client used to retrieve Secrets</param>
+        public DaprSecretStoreConfigurationProvider(string store, IEnumerable<DaprSecretDescriptor> secretDescriptors, DaprClient client)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(store, nameof(store));
+            ArgumentVerifier.ThrowIfNull(secretDescriptors, nameof(secretDescriptors));
+            ArgumentVerifier.ThrowIfNull(client, nameof(client));
+
+            this.store = store;
+            this.secretDescriptors = secretDescriptors;
+            this.client = client;
+        }
+
+        public override void Load() => LoadAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task LoadAsync()
+        {
+            var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var secretDescriptor in secretDescriptors)
+            {
+                var result = await client.GetSecretAsync(store, secretDescriptor.SecretName, secretDescriptor.Metadata).ConfigureAwait(false);
+
+                foreach (var key in result.Keys)
+                {
+                    if (data.ContainsKey(key))
+                    {
+                        throw new InvalidOperationException($"A duplicate key '{key}' was found in the secret store '{store}'. Please remove any duplicates from your secret store.");
+                    }
+
+                    data.Add(key, result[key]);
+                }
+            }
+
+            Data = data;
+        }
+
+        private void WaitForDapr()
```

Comment:
Hi,

_Disclaimer: first time reviewing PR in this repository, so forgive me if I am out of line here._

Method seems to be unused, does this method have a purpose to be used in the future, or was it included during development?

Reply:
Hey thanks! Somehow I forgot to remove it.